shared/bit.go

```diff
@@ -0,0 +1,50 @@
+package shared
+
+import (
+	"math"
+)
+
+// CheckBit checks if a bit in a bit field is one.
+func CheckBit(bitfield []byte, index int) bool {
+	chunkLocation := (index + 1) / 8
+	indexLocation := (index + 1) % 8
+	if indexLocation == 0 {
+		indexLocation = 8
+	} else {
+		chunkLocation++
+	}
+
+	field := bitfield[chunkLocation-1] >> (8 - uint(indexLocation))
+	return field%2 != 0
+}
+
+// BitSetCount counts the number of 1s in a byte using the following algo:
+// https://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetParallel
+func BitSetCount(bytes []byte) int {
```

Comment:
Can we use this library for this? 

https://github.com/steakknife/hamming/blob/16b83346f022ed8fdfc17798a88c44da64b9010d/popcount.go#L60

They have an optimized lookup table which claims to be significantly faster: 

```
BenchmarkCountBitsByte-4            	2000000000	         0.75 ns/op
BenchmarkCountBitsByteAlt-4         	1000000000	         2.37 ns/op
```

`BenchmarkCountBitsByteAlt` is the same implementation as this method, i believe.

Reply:
Good idea. Done!